getSigmaPointApp.asv

```diff
@@ -0,0 +1,373 @@
+%    .sigma ... @sigma(phi) defines the parametrisation of the technical
+%            noise
+%    .dsigmadphi ... @dsigmadphi(phi) defines the derivative of sigma with
+%            respect to phi
+% op_SP  ... struct containing options on what the output struct should
+%            contain
+%    .nderiv ... flag indicating whether derivatives should be computed
+%    .req ... vector of flags indicating whether a certain output is
+%            requested
+%            (1) my
+%            (2) Cy
+%            (3) Cxy
+%            (4) mz
+%            (5) Cz
+%    .type_D ... string specifying the parametrisation of the covariance
+%                matrix for the random effects. either
+%            'diag-matrix-logarithm' for diagonal matrix with log. parametrisation or
+%            'matrix-logarithm' for full matrix with log. parametrisation
+%    .approx ... string specifying the approximation method. either
+%            'sp' for sigma_points
+%            'sampled' for sampling based approximation
+%            'dmd' for approximation by dirac mixtrue distribution
+%    .n_samples ... number of dirac mixture components, only defined if op_SP.approx='dmd'
+%
+% OUTPUTS:
+% =======
+% whether a certain output is requested must be specified in op_SP.req
+% SP.my     ... mean of the sigma-point approximation
+%            [time x observables]
+% SP.Cy     ... covariance of the sigma-point approximation
+%            [time x observables x observables]
+% SP.Cxy    ... temporal cross-correlation of the sigma-point approximation
+%            [time x time x observables x observables]
+% SP.mz     ... full state vector (all observables at all times)
+%            [time*observables]
+% SP.Cz     ... full state covariance, this contains both covariances across
+%            time and states
+%            [time*observables x time*observables]
+% SP.B_SP   ... locations of employed sigma points
+%            [parameters x 2*parameters+1]
+% SP.Y      ... evaluation of nonfun at the employed sigma points
+%            [time x observables x 2*parameters+1]
+%
+% OPTIONAL OUTPUTS (only computed when derivatives are computed):
+% =======
+% SP.dmydxi ... derivative of my w.r.t. xi       
+%            [time x observables x parameters]
+% SP.dCydxi ... derivative of Cy w.r.t. xi   
+%            [time x observables x observables x parameters]
+% SP.dCxydxi... derivative of Cxy w.r.t. xi
+%            [time x time x observables x observables x parameters]
+% SP.dmzdxi ... derivative of mz w.r.t. xi 
+%            [time*observables x parameters]
+% SP.dCzdxi ... derivative of Cz w.r.t. xi 
+%            [time*observables x time*observables x parameters]
+% SP.dB_SPdxi... derivative of B_SP w.r.t. xi 
+%            [parameters x 2*parameters+1 x parameters]
+% SP.dYdxi  ... derivative of Y w.r.t. xi 
+%            [time x observables x 2*parameters+1 x parameters]
+
+function SP = getSigmaPointApp(varargin)
+
```

Comment:
What does this function/file actually do? Do we really need it, or is it just here due to some name doubling? I would somehow guess the latter one... If so, it can be deleted...